### ISSUE TITLE
The program runs too slowly

### ISSUE THREAD

**Issue by wushangbin** [YerevaNN/mimic3-benchmarks#107]
Hello, when I run the "python -um mimic3models.in_hospital_mortality.main --network mimic3models/keras_models/lstm.py --dim 16 --timestep 1.0 --depth 2 --dropout 0.3 --mode train --batch_size 8 --output_dir mimic3models/in_hospital_mortality", The program runs too slowly, an epoch may take more than an hour, and when the program is running, the utilization of the gpu is very low. May I ask if this is normal？
Thank you very much!!!

**Comment by wayne0320** [comment]
> a

hello,may I know the version of python and extension package you were using？

**Comment by alanflanders** [comment]
I implemented multi-processing on the data generator which has sped up the process a great deal. Have also implemented other performance improvements. I see there are pending pull requests. If this project is still active please reach out and I can submit a pull request. Note that some changes were also required to make the benchmark code compatible with Tensorflow 2.0. Also note that the performance improvements has broken the deep supervision model type.

**Comment by tingwei161803** [comment]
I had the same problem. The code seems to train on CPU, not GPU.
the requirement package contain tensorflow==1.15.0, which is a CPU version
How do I switch to a newer version of TensorFlow or use a GPU?
